src/ring_theory/localization.lean

```diff
@@ -703,6 +715,43 @@ variables (f : localization_map M S)
 
 section ideals
 
+/-- Explicit definition of the ideal given by `ideal.map f.to_map I` -/
+def to_map_ideal (I : ideal R) : ideal S :=
```

Comment:
This is quite strange. Would it be better to just give a lemma saying `(ideal.map f.to_map I : set R) = ` { z : S | ∃ x : I × M, z * (f.to_map x.2) = f.to_map x.1}`?

Reply:
The only reason I created this explicit definition as an ideal was so that when proving equality between the carriers I could rely on theorems about ideals. I could just inline the proofs of those statements as they come up instead, rather than create this whole definition (Or just have this definition be local inside of the proof of the statement).

Reply:
This seems like a sensible reason, but maybe add to the docstring saying this is a definition that isn't intended to be used outside this file. You could even make this theorem and the theorems that mention it `private`, and write a lemma `mem_map_to_map_iff` or something, giving the equality of sets.

Reply:
Made this private and added a more better explanation in the docstring, and exposed a public theorem that just gives the equality of sets that would be used in practice.